src/test/java/be/seeseemelk/mockbukkit/block/state/BlockStateMockTest.java

```diff
@@ -94,4 +91,36 @@ public void testUpdateForce()
 		assertTrue(chest.update(true));
 		assertTrue(block.getState() instanceof Chest);
 	}
+
+	// Tests that different instances of the placed block state of a block, both with the same material, are equal
+	@Test
+	public void testEquals() {
+		Block block = new BlockMock(Material.STONE, new Location(null, 0, 64, 0));
+
+		assertEquals(block.getState(), block.getState());
+	}
+
+	// Tests that different instances of the placed block state of a block, with different materials, are unequal
+	// Also tests that block states with the same material, but with different parent blocks are unequal
+	@Test
+	public void testNotEquals() {
+		Block blockA = new BlockMock(Material.STONE, new Location(null, 0, 64, 0));
+		Block blockB = new BlockMock(Material.AIR, new Location(null, 0, 65, 0));
+		Block blockC = new BlockMock(Material.STONE, new Location(null, 0, 65, 0));
+
+		assertNotEquals(blockA.getState(), blockB.getState());
```

Comment:
Here you compare if two blocks with different type and different parents are equal. Did you mean to compare `blockA` and `blockC` here, as you are already comparing two blocks with different types on the next line.

Reply:
Correct yes, thanks for pointing this out.